src/Features/LensEffects.cpp

```diff
@@ -0,0 +1,1110 @@
+	auto skyProperty = reinterpret_cast<RE::BSSkyShaderProperty*>(Pass->shaderProperty);
+
+	if (skyProperty->uiSkyObjectType == RE::BSSkyShaderProperty::SkyObject::SO_SUN_GLARE) {
+		lens.overrideShader = true;
+		lens.shaderdesc = Shaders::Bypass;
+	}
+
+	else if (skyProperty->uiSkyObjectType == RE::BSSkyShaderProperty::SkyObject::SO_ATMOSPHERE) {
+		if (Pass->passEnum == 0x5C00005E && RenderFlags == 65) {
+			lens.overrideShader = true;
+			lens.shaderdesc = Shaders::AttachLUT;
+		}
+	}
+
+	else if (skyProperty->uiSkyObjectType == RE::BSSkyShaderProperty::SkyObject::SO_CLOUDS) {
+		if ((Pass->passEnum == 0x5C000062 || Pass->passEnum == 0x5C000063) && RenderFlags == 65) {
+			lens.overrideShader = true;
+			lens.shaderdesc = Shaders::AttachLUT;
+			lens.useCloudLUT = true;
+		}
+	}
+
+	else if (skyProperty->uiSkyObjectType == RE::BSSkyShaderProperty::SkyObject::SO_SUN) {
+		if (lens.SunScale == 0.0f && Pass->passEnum == 0x5C000061 && (RenderFlags == 65 || RenderFlags == 8)) {
+			D3D11_TEXTURE2D_DESC sunDesc = {};
+			skyProperty->pBaseTexture->rendererTexture->texture->GetDesc(&sunDesc);
+			lens.SunScale = 24.0f / sunDesc.Width;
+		}
+	}
+
+	func(This, Pass, RenderFlags);
+}
+
+void LensEffects::Hooks::LensFlare_CheckResources::thunk()
+{
+	auto& lens = globals::features::lensEffects;
+
+	lens.renderdata->CheckRefData();
+
+	if (!*lens.skyrim_FlareData)
+		*lens.skyrim_FlareData = reinterpret_cast<uintptr_t>(lens.renderdata);
+}
+
+void LensEffects::Hooks::LensFlareVisibility_CheckRenderCondition::thunk(RE::NiCamera* camera, void* shader)
+{
+	auto& lens = globals::features::lensEffects;
+
+	func(camera, shader);  //set skyrim_RunFlarePtr
+
+	bool sunVisble = static_cast<bool>(*lens.skyrim_RunFlarePtr);
+
+	if (lens.disableSunFX)
+		sunVisble = false;
+
+	if (*lens.skyrim_FlareData) {
+		lens.renderdata->ResetEffects(sunVisble);
+		*lens.skyrim_RunFlarePtr = 1;
+	} else
+		*lens.skyrim_RunFlarePtr = 0;
+}
+
+void LensEffects::Hooks::BSImagespaceShader_Render<RE::ImageSpaceManager::ISLensFlareVisibility>::thunk(void* shader, RE::BSTriShape* shape, RE::ImageSpaceEffectParam* param)
+{
+	auto& lens = globals::features::lensEffects;
+
+	lens.overrideShader = true;
+	lens.shaderdesc = Shaders::OcculsionMask;
+
+	func(shader, shape, param);
+}
+
+void LensEffects::Hooks::BSImagespaceShader_Render<RE::ImageSpaceManager::ISLensFlare>::thunk(void* shader, RE::BSTriShape* shape, RE::ImageSpaceEffectParam* param)
+{
+	auto& lens = globals::features::lensEffects;
+
+	lens.overrideShader = true;
+	lens.shaderdesc = lens.renderdata->UpdateCurrentEffect();
+
+	if (lens.renderdata->GetCurrentEffect().IsWeatherShader() && lens.shaderdesc != Shaders::Bypass)
+		lens.GetWeatherShader();
+
+	func(shader, shape, param);
+}
+
+void LensEffects::DrawSettings()
+{
+	static auto& mainSettings = settings.mainsettings;
+	static auto& coldSettings = settings.coldsettings;
+
+	if (ImGui::Button("Save Preset"))
+		SaveCustomSettings(settings);
```

Comment:
You shouldn't need a button to save settings. Just have a function def to handle serialization when the main save is called.